src/App.tsx

```diff
@@ -1,34 +1,86 @@
 /* eslint-disable max-len */
-import React from 'react';
+
+// default imports
+
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+// components
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+// outer function
+
+import { getTodos } from './api';
+import { filterTodos } from './utils/filterTodos';
+
+// types
+
+import { Todo } from './types/Todo';
+import { FilterByState } from './types/FilterByState';
+
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo>(null);
```

Comment:
The `useState` hook for `selectedTodo` is initialized with `null`, which is correct. However, ensure that the `setSelectedTodo` function in the `TodoList` and `TodoModal` components is typed to accept a `Todo` object or `null`.